cogs/moderation.py

```diff
@@ -51,11 +79,72 @@ async def ban(self, ctx: Context, member: discord.Member, *, reason: str = "No R
         await member.send(embed=embed)
         await member.ban(reason=reason)
 
+    class ActionReason(Converter):
+        async def convert(self, ctx: Context, argument: str) -> str:
+            ret = f"{ctx.author} (ID: {ctx.author.id}): {argument}"
+
+            if len(ret) > 512:
+                reason_max = 512 - len(ret) + len(argument)
+                raise BadArgument(f"Reason is too long ({len(argument)}/{reason_max})")
+            return ret
+
+    class MemberID(Converter):
+        async def convert(self, ctx: Context, argument: str) -> t.Union[str, type]:
+            try:
+                member = await MemberConverter().convert(ctx, argument)
+            except BadArgument:
+                try:
+                    member_id = int(argument, base=10)
+                    member = await resolve_member(ctx.guild, member_id)
```

Comment:
```suggestion
        async def convert(self, ctx: Context, member_id: int) -> t.Union[str, type]:
            try:
                member = await MemberConverter.convert(ctx, argument)
            except BadArgument:
                try:
                    member = await resolve_member(ctx.guild, member_id)
```
We can simply accept `member_id` as `int` straight away, no need to convert it from string.
In case you'd really want to convert it from `str`, you also shouldn't specify `base=10` as that is the default value anyway.